Shopify/cli
# Fix an issue in `shopify theme dev` that was affecting asset loading on local servers (2)

### WHY are these changes introduced?

This PR fixes a corner case related to https://github.com/Shopify/cli/issues/2094

### WHAT is this pull request doing?

When the `storefront_vanity_assets_url` flag is activated, assets URL get a different shape. So, this PR modifies the regex for the assets and fonts middlewares in the `shopify theme dev` commands [1]. This change allows images to load directly from the disk when users access the local server (`http://127.0.0.1:9292`), rather than pointing to the CDN URLs.

[1] This flag doesn't affect theme app extensions.

### How to test your changes?

- Activate the `storefront_vanity_assets_url` flag
- Run `shopify theme init`
- Open the theme directory
- Run `shopify theme dev`
- Notice that images and fonts load from the local server now

**Before**
![Screenshot 2023-06-07 at 12 27 20](https://github.com/Shopify/cli/assets/1079279/7c43eb8a-2377-4fcd-8553-f5179a1152c1)

**After**
![Screenshot 2023-06-07 at 12 32 02](https://github.com/Shopify/cli/assets/1079279/dc3febbf-15f9-44ec-a2a4-fc58a15a244f)


### Post-release steps

None.

### Measuring impact

How do we know this change was effective? Please choose one:


### Checklist
